src/particle/corsika/corsika7id.py

```diff
@@ -0,0 +1,168 @@
+    ) -> tuple[Self, bool]:
+        """
+        Constructor from the particle description returned by Corsika7
+        in the particle data sub-block, mother particle data sub-block or
+        the grandmother particle data sub-block.
+
+        Returns
+        -------
+        Corsika7ID: The Corsika7 id
+        bool:       If the particle is a (grand)motherparticle.
+        """
+        cid = abs(particle_description) // 1000
+        ismother = particle_description < 0
+
+        if cls._is_non_particle_id(cid):
+            return cls(cid), ismother
+
+        # this catches the case, of nuclei with no known PDGid
+        if cid >= 200 and cid < 5699:
+            return cls(cid), ismother
+
+        if cid in _bimap:
+            return cls(cid), ismother
+
+        raise MatchingIDNotFound(
+            f"Non-existent Corsika7ID for particle_description {particle_description}!"
+        )
+
+    @classmethod
+    def _is_non_particle_id(cls: type[Self], id: int) -> bool:
+        """
+        returns True if the id is a valid id, but not a particle, False otherwise.
+        """
+        return id in _non_particles or id // 1000 == 8888 or id == 9900
+
+    def is_particle(self) -> bool:
+        """
+        Returns if the corsikaid really belongs to a particle, since some are for example additional information.
+        """
+        iid = int(self)
+
+        return not self._is_non_particle_id(iid)
+
+    def name(self) -> str:
+        """
+        Returns a human readable name of the Corsika ID.
+        This also works for non-particles (is_particle()==false).
+
+        Raises
+        ------
+        ParticleNotFound
+            If it is a 'valid' PDG particle, but unknown.
+            This for example happens with strange nuclei, which are not in the nuclei list.
+        """
+        from ..particle.particle import Particle
+
+        if self.is_particle():
+            return str(Particle.from_pdgid(self.to_pdgid()).name)
+
+        iid = int(self)
+
+        if iid in _non_particles:
+            return _non_particles[iid]
+
+        if iid // 1000 == 8888:
+            return "weights of preceding particle (MULTITHIN option)"
+
+        if iid == 9900:
+            return "Cherenkov photons on particle output file"
+
+        raise RuntimeError("This should be unreachable.")
+
+    def to_pdgid(self) -> PDGID:
+        """
+        Raises
+        ------
+        InvalidParticle
+            If it is a valid Corsika particle, but not a valid PDGid particle.
+        """
+        from ..particle.particle import InvalidParticle
+
+        if self not in _bimap:
+            raise InvalidParticle(
+                f"The Corsika7Id {self} is not a valid PDGID particle."
+            )
+        return PDGID(_bimap[self])
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}: {int(self):d}>"
+
+    def __str__(self) -> str:
+        return repr(self)
```

Comment:
Apologies if the answer is trivial - any reason not to implement `__invert__` and `__neg__` as for other ID converter classes?

Reply:
Actually: Yes. Corsika7 is very - let's say - interesting, regarding negative IDs. There is a special option to compile Corsika7 in a way, where they print out information about the (grand)mother particle for each arriving particle. They get negative IDs. So a negative ID is somewhat special in Corsika: It refers to a property of the particle, not actually the particle type. I think it's best not to implement any type of arithmetic on the Corsika7 IDs... 
But if there are scenarios where this can be useful, it wouldn't be too confusing to add it back...

Reply:
OK, I see. Such info could maybe be added in a short form to the class docstring so that anyone looking at some point won't be surprised. Just a tought ... I tend to work like that to allow me to forget ;-).